tests/torchtune/modules/loss/test_liger_ce_loss.py

```diff
@@ -0,0 +1,205 @@
+
+        return fused_loss, standard_loss
+
+    @gpu_test(gpu_count=1)
+    @pytest.mark.parametrize("compile", [False, True])
+    def test_liger_fused_cross_entropy(self, compile):
+        """Test gradient flow through full forward/backward pass with optimizer step"""
+        # Create model for fused loss
+        fused_model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(fused_model, min_val=-0.1, max_val=0.1)
+
+        # Create model for standard loss
+        model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(model, min_val=-0.1, max_val=0.1)
+
+        fused_loss, standard_loss = self.compute_loss(
+            fused_model, model, "cuda", compile
+        )
+
+        # Verify:
+        # 1. Validate the results are close enough
+        assert_expected(fused_loss, standard_loss, rtol=1e-2, atol=1e-2)
+
+        # 2. Validate the weight updates are close enough
+        for p1, p2 in zip(fused_model.parameters(), model.parameters()):
+            assert_expected(p1, p2, rtol=1e-2, atol=1e-2)
+
+    def loss_worker(self, rank, compile, port):
+        os.environ["MASTER_ADDR"] = "localhost"
+        os.environ["MASTER_PORT"] = str(port)
+        device = torch.device(f"cuda:{rank}")
+        dist.init_process_group(
+            backend="nccl", rank=rank, world_size=WORLD_SIZE, device_id=device
+        )
+        torch.cuda.set_device(device)
+        mesh = ParallelDims(
+            dp_replicate=1, dp_shard=WORLD_SIZE // 2, tp=2, cp=1, world_size=WORLD_SIZE
+        ).build_mesh("cuda")
+        plan = {
+            "output": ColwiseParallel(
+                input_layouts=Replicate(), output_layouts=Replicate()
+            )
+        }
+
+        # Create model for fused loss
+        fused_model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(fused_model, min_val=-0.1, max_val=0.1)
+        parallelize_module(fused_model, mesh["tp"], parallelize_plan=plan)
+        for m in [fused_model.layer, fused_model.output, fused_model]:
+            fully_shard(m, mesh=mesh["dp_shard"])
+
+        # Create model for standard loss
+        model = Model(VOCAB_SIZE, EMBED_DIM).cuda()
+        fixed_init_model(model, min_val=-0.1, max_val=0.1)
+        parallelize_module(model, mesh["tp"], parallelize_plan=plan)
+        for m in [model.layer, model.output, model]:
+            fully_shard(m, mesh=mesh["dp_shard"])
+
+        fused_loss, standard_loss = self.compute_loss(
+            fused_model, model, device, compile, mesh["dp_shard"].get_local_rank()
+        )
+
+        # Verify:
+        # 1. Validate the results are close enough
+        assert_expected(fused_loss, standard_loss, rtol=1e-2, atol=1e-2)
+
+        # 2. Validate the weight updates are close enough
+        for p1, p2 in zip(fused_model.parameters(), model.parameters()):
+            assert_expected(p1, p2, rtol=1e-2, atol=1e-2)
+
+        dist.destroy_process_group()
+
+    @gpu_test(gpu_count=WORLD_SIZE)
```

Comment:
nit: just say 4 explicitly here. I like to be able to look directly at the test and see how many GPUs it needs